Source/DafnyLanguageServer.Test/Lookup/DefinitionTest.cs

```diff
@@ -424,5 +424,44 @@ class Test {
       Assert.AreEqual(DocumentUri.FromFileSystemPath(Path.Combine(Directory.GetCurrentDirectory(), "Lookup/TestFiles/foreign.dfy")), location.Uri);
       Assert.AreEqual(new Range((3, 2), (3, 13)), location.Range);
     }
+
+    [TestMethod]
+    public async Task Refinement() {
+      var source = @"
+module {>0:A<} {
+  class X { }
+  class T {
+    method M(x: int) returns (y: int)
+      requires 0 <= x;
+      ensures 0 <= y;
+    {
+      y := 2 * x;
+    }
+    method Q() returns (q: int, r: int, {>1:s<}: int)
+      ensures 0 <= q && 0 <= r && 0 <= s;
+    {  // error: failure to establish postcondition about q
+      r, s := 100, 200;
+    }
+  }
+}
+
+module B refines ><A {
+  class C { }
+  datatype Dt = Ax | Bx
+  class T ... {
+    method P() returns (p: int)
+    {
+      p := 18;
+    }
+    method M(x: int) returns (y: int)
+      ensures y % 2 == 0;  // add a postcondition
+    method Q ...
+      ensures 12 <= r;
+      ensures 1200 <= ><s;  // error: postcondition is not established by
```

Comment:
This syntax is excellent to understand what the test is about. Again, thanks for having implemented it !